packages/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/MarkersController.java

```diff
@@ -0,0 +1,115 @@
+package io.flutter.plugins.googlemaps;
+
+import com.google.android.gms.maps.GoogleMap;
+import com.google.android.gms.maps.model.Marker;
+import com.google.android.gms.maps.model.MarkerOptions;
+import io.flutter.plugin.common.MethodChannel;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+class MarkersController {
+
+  private final Map<MarkerId, MarkerController> markerIdToController;
+  private final Map<String, MarkerId> googleMapsMarkerIdToDartMarkerId;
+  private final MethodChannel methodChannel;
+  private GoogleMap googleMap;
+
+  MarkersController(MethodChannel methodChannel) {
+    this.markerIdToController = new HashMap<>();
+    this.googleMapsMarkerIdToDartMarkerId = new HashMap<>();
+    this.methodChannel = methodChannel;
+  }
+
+  void setGoogleMap(GoogleMap googleMap) {
+    this.googleMap = googleMap;
+  }
+
+  void addMarkers(List<Object> markersToAdd) {
+    if (markersToAdd != null) {
+      for (Object markerToAdd : markersToAdd) {
+        addMarker(markerToAdd);
+      }
+    }
+  }
+
+  void changeMarkers(List<Object> markersToChange) {
+    if (markersToChange != null) {
+      for (Object markerToChange : markersToChange) {
+        changeMarker(markerToChange);
+      }
+    }
+  }
+
+  void removeMarkers(List<Object> markerIdsToRemove) {
+    if (markerIdsToRemove == null) {
+      return;
+    }
+    for (Object rawMarkerId : markerIdsToRemove) {
+      if (rawMarkerId != null) {
```

Comment:
uber nit: my personal preference would be to reverse the condition and `continue` to save one nesting level. Sending you some better articulated explanation offline.